Desktop/html/js/jaspwidgets.js

```diff
@@ -1,5 +1,26 @@
 var JASPWidgets = {};
 
+var i18nStrs = null;
+// The global translation function needs to be loaded before all other .js files.
+window.setI18nStrings = function (i18nObject) {
+        i18nStrs = i18nObject
+        return i18nStrs
+}
+
+// Use i18n(...) to call strings to be translated then update them in MainPage.qml
+function i18n(str) {
+	if (typeof str === "string" && str instanceof String) {
+		console.log("i18n can only be used for literal strings")
+	} else {
+		let findI18nStr = i18nStrs[str] !== null && i18nStrs[str] !== undefined;
```

Comment:
Why don't you use the hasOwnProperty method? 
I'm not sure, but I'm afraid by testing ` i18nStrs[str] !== null` this will insert automatically an entry in the object if the key does not exist.

Reply:
yes you are right, that should be more robust.

Reply:
But this still cannot avoid returning `null `and `undefined`, because this is still valid for undefined